operators/constellation-node-operator/controllers/nodeversion_controller.go

```diff
@@ -85,6 +94,17 @@ func (r *NodeVersionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	if err := r.Get(ctx, req.NamespacedName, &desiredNodeVersion); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
+	// Check if we need to upgrade the cluster version.
+	serverVer, err := r.ServerVersion()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// GitVersion is the semantic version of the Kubernetes server e.g. "v1.24.9"
+	if serverVer.GitVersion != desiredNodeVersion.Spec.KubernetesClusterVersion {
+		r.tryStartClusterVersionUpgrade(ctx, req.NamespacedName)
+	}
```

Comment:
Not entirely convinced if this is necessary, but maybe this comparison should be less strict by either using [strings.EqualFold](https://pkg.go.dev/strings#EqualFold) or semver compare?
I'm concerned about cases where the `GitVersion` field uses slightly different representation of the same semantic version which would prevent this from ever reaching the desired state.